### PR TITLE
[test] test whether the RCP supports the Link Metrics

### PR DESCRIPTION
**Pull request by zhanglongxia**
This commit adds a test case to `cp-tests` to test whether the RCP supports the Link Metrics.

### CODE REVIEW

File: tools/cp-caps/rcp_caps_test.py [hunk 1]
```diff
@@ -136,9 +137,67 @@ def test_throughput(self):
 
         self.__output_format_string('Throughput', self.__bitrate_to_string(results['receiver']['bitrate']))
 
+    def test_link_metrics(self):
+        """Test whether the DUT supports Link Metrics Initiator and Subjector."""
```
Comment (Irving-cl) [inline tools/cp-caps/rcp_caps_test.py:141]:
```suggestion
        """Test whether the DUT supports Link Metrics Initiator and Subject."""
```

File: tools/cp-caps/rcp_caps_test.py [hunk 2]
```diff
@@ -136,9 +137,67 @@ def test_throughput(self):
...
+        self.__dut.leave()
+
     #
     # Private methods
     #
+    def __run_link_metrics_test_commands(self, initiator: OTCI, subject_address: Ip6Addr) -> bool:
+        seriesid = 1
+        series_flags = 'ldra'
+        link_metrics_flags = 'qr'
+        probe_lenght = 10
```
Comment (Irving-cl) [inline tools/cp-caps/rcp_caps_test.py:173]:
```suggestion
        probe_length = 10
```

File: tools/cp-caps/rcp_caps_test.py [hunk 3]
```diff
@@ -136,9 +137,67 @@ def test_throughput(self):
...
+
+        ret = initiator.linkmetrics_config_enhanced_ack_register(subject_address, link_metrics_flags)
+        if not ret:
+            return False
+
+        ret = initiator.linkmetrics_config_forward(subject_address, seriesid, series_flags, link_metrics_flags)
+        if not ret:
+            return False
+
+        initiator.linkmetrics_probe(subject_address, seriesid, probe_lenght)
```
Comment (Irving-cl) [inline tools/cp-caps/rcp_caps_test.py:183]:
```suggestion
        initiator.linkmetrics_probe(subject_address, seriesid, probe_length)
```

File: tools/cp-caps/rcp_caps_test.py [hunk 4]
```diff
@@ -136,9 +137,67 @@ def test_throughput(self):
...
+
+        results = initiator.linkmetrics_request_single(subject_address, link_metrics_flags)
+        ret = 'lqi' in results.keys() and 'rssi' in results.keys()
+        if not ret:
+            return False
```
Comment (Irving-cl) [inline tools/cp-caps/rcp_caps_test.py:188]:
```suggestion
        if not ('lqi' in results.keys() and 'rssi' in results.keys()):
          return False
```
I feel `ret` is not necessary in this function and all the three lines blocks can be changed to two lines.

**Review by Irving-cl — APPROVED** [review]
LGTM.

A few nit suggestions.

**Review by abtink — APPROVED** [review]
👍

File: tools/cp-caps/rcp_caps_test.py [hunk 5]
```diff
@@ -136,9 +137,62 @@ def test_throughput(self):
...
+        if not initiator.linkmetrics_config_enhanced_ack_register(subject_address, link_metrics_flags)
+            return False
+
+        if not initiator.linkmetrics_config_forward(subject_address, seriesid, series_flags, link_metrics_flags)
+            return False
+
+        initiator.linkmetrics_probe(subject_address, seriesid, probe_length)
+
+        results = initiator.linkmetrics_request_single(subject_address, link_metrics_flags)
+        if not 'lqi' in results.keys() and 'rssi' in results.keys()
```
Comment (superwhd) [inline tools/cp-caps/rcp_caps_test.py:184]: What's the precedence of `not` and `and`? Maybe add parentheses to make the condition easier to understand.

**Review by superwhd — APPROVED** [review]
LGTM 👍

File: tools/cp-caps/rcp_caps_test.py (hunk 5 above)
Comment (zhanglongxia) [reply tools/cp-caps/rcp_caps_test.py]: I have updated the code to add `()` to include `and` conditions.

**Review by jwhui — APPROVED** [review]

**Pull request merged**